### PR TITLE
Suggest profiles when no auth config; clean up legacy keyring entries

### LINKED ISSUE #2187: Auth realm not considered when using --all option
### Describe the Bug

If I run a plan for a component `atmos terraform plan <component> -s <stack>`, credentials are stored into `.config/atmos/<realm>`

Instead, if I run a plan for the entire stack `atmos terraform plan -s <stack> --all`, I get the following warning:

`WARN  Credentials found under realm "b1fe90cf620cb71a" but current realm is "(no realm)". This typically happens after changing auth.realm in config or ATMOS_AUTH_REALM. Run 'atmos auth login' to re-authenticate under the current realm.`

Then looking into the `.config/atmos/` directory I can see two different folders
```
ls -l atmos/
drwx------ 3 root root 4096 Mar 13 10:30 aws
drwx------ 3 root root 4096 Mar 13 10:30 b1fe90cf620cb71a
```

It seems that with `--all` credentials are stored without the realm in the path (I actually haven’t been asked to reauthenticate)

### Expected Behavior

Even with the `--all` option auth realm is used in the path for storing credentials

### Steps to Reproduce

_No response_

### Screenshots

_No response_

### Environment

- Atmos version 1.209.0
- AWS SSO as provider

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by osterman**
## what

- **Fix: Propagate auth realm through `CopyGlobalAuthConfig` and `buildGlobalAuthSection`** so that `ExecuteTerraform` (both single-component and `--all` paths) uses the correct realm for credential storage. This is the root cause fix for #2187.
  - `CopyGlobalAuthConfig` in `pkg/auth/config_helpers.go` was missing `Realm` and `RealmSource` fields — when `--all` iterates components, each call to `CopyGlobalAuthConfig` produced an auth config with an empty realm
  - `buildGlobalAuthSection` in `internal/exec/utils_auth.go` was not including `realm` in the auth map — the deep-merge path lost the realm value
- When running `atmos auth list` with no providers/identities configured, Atmos now checks if profiles exist and suggests using the `--profile` flag to load auth configuration from profile-specific atmos.yaml files
- After successfully caching credentials under the current realm, legacy (pre-realm) keyring entries are automatically cleaned up to prevent realm mismatch warnings
- Uses the error builder pattern with `ErrAuthNotConfigured` sentinel error for rich, helpful error messages
- Improved error message when identity `kind` is empty (not configured in active profile)

## why

When running `atmos terraform plan -s <stack> --all`, credentials were stored without the auth realm in the path, causing realm mismatch warnings. Single-component runs (`atmos terraform plan <component> -s <stack>`) worked correctly because they used a different code path that preserved the realm. The two-line root cause fix ensures realm propagation through both paths.

Users working with profile-based configurations (like infra-live) need better guidance when auth config is not in the base atmos.yaml. Suggesting available profiles reduces friction and helps users discover the correct workflow.

The keyring cleanup prevents accumulated migration warnings when users upgrade to realm-aware credential management and transition between different credential storage locations.

## testing

- Unit tests verify `Realm`/`RealmSource` are copied in `CopyGlobalAuthConfig` (including deep-copy mutation safety)
- Unit tests verify `realm` is included/excluded in `buildGlobalAuthSection` based on whether it's set
- Integration tests verify realm propagation through `getMergedAuthConfigWithFetcher` (the `--all` code path)
- Tests confirm no-realm configurations continue to work correctly (backward compatibility)
- Tests for profile suggestion, legacy credential cleanup, and empty identity kind error messages
- Updated golden snapshots reflecting realm now appearing in `describe component` output

## references

- Addresses the use case where auth providers/identities live in configuration profiles rather than the base atmos.yaml
- Closes #2187

## Summary by CodeRabbit

* **New Features**
  * Auth list now suggests available profiles and shows a --profile usage hint when no auth providers or identities are configured.

* **Improvements**
  * Automatic cleanup of legacy credential files and keyring entries after successful authentication to prevent realm-mismatch warnings.
  * Auth configuration now preserves and exposes a realm value where applicable; CLI describe output snapshots updated.

* **Bug Fixes**
  * Clearer, actionable error when an identity is missing its kind, including guidance to list/select profiles.

* **Tests**
  * Expanded tests for profile suggestions, realm propagation, merging, and legacy cleanup edge cases.

### CODE REVIEW

File: pkg/auth/manager.go [hunk 1]
```diff
@@ -231,6 +231,14 @@ func (m *manager) Authenticate(ctx context.Context, identityName string) (*types
 	m.chain = chain
 	log.Debug("Authentication chain discovered", logKeyIdentity, identityName, "chainLength", len(chain), "chain", chain)
 
+	// Clean up legacy (pre-realm) keyring and file entries to prevent realm mismatch warnings.
+	// This runs early in the Authenticate flow so it executes even when cached credentials
+	// allow the provider authentication step to be skipped.
+	m.deleteLegacyCredentialFiles()
+	for _, step := range chain {
+		m.deleteLegacyKeyringEntry(step)
+	}
```
[the remark this answers is not in the document]
Comment (osterman) [reply pkg/auth/manager.go]: Good catch! Updated both docstrings in `realm_mismatch.go` to accurately reflect that `deleteLegacyKeyringEntry` and `deleteLegacyCredentialFiles` run during the early cleanup phase of `Authenticate`, before `authenticateChain`, and remove legacy entries unconditionally.

**Comment by aknysh** [comment]
@coderabbitai full review please.
Also check if this PR fixes the issues reported in https://github.com/cloudposse/atmos/issues/2187

**Comment by aknysh** [comment]
@coderabbitai please check if this PR fixes the issues reported in https://github.com/cloudposse/atmos/issues/2187
